app/components/Modals/SendModal/SendModal.jsx

```diff
@@ -110,19 +110,25 @@ export default class SendModal extends Component<Props, State> {
   handleConfirmAddRecipient = (entry: SendEntryType) => {
     const { showErrorNotification } = this.props
     const { balances } = this.state
-    const error = validateTransactionBeforeSending(balances[entry.symbol], entry)
 
-    if (error) {
-      showErrorNotification({ message: error })
-    } else {
-      const newBalance = toBigNumber(balances[entry.symbol]).minus(entry.amount).toString()
-
-      this.setState({
-        entries: [...this.state.entries, entry],
-        balances: { ...balances, [entry.symbol]: newBalance },
-        display: DISPLAY_MODES.CONFIRM
-      })
-    }
+    isInBlacklist(entry.address).then(inBlacklist => {
```

Comment:
FYI using async/await syntax here can remove a layer of nesting and improve the readability.  We've been favoring this syntax over raw promises throughout the codebase.

You would first need to change the method signature to include `async`, i.e.:

```js
handleConfirmAddRecipient = async (entry: SendEntryType) => { ... }
```

Then the skeleton of this function simply becomes:

```js
if (await isInBlacklist(entry.address)) {
  ...
} else {
  ...
}
```

Reply:
Unfortunately, making that function async/await removes access to the "this" variable. I get ``Cannot read property 'showErrorNotification' of undefined`` or ``ReferenceError: _this11 is not defined``. There's a weird workaround to this in withAddressCheck.jsx, but applying it breaks the render function. I'm going to leave it up to you what to do with this function and commit the other changes in the meantime. Let me know what I should do with this.

Reply:
Ahh, that should be fixable by using `async function(...) { ... }` instead of `async (...) => { ... }`.  Regardless, I can probably merge this as-is and look into it afterwards.